base/linalg/tridiag.jl

```diff
@@ -559,15 +559,18 @@ function getindex{T}(A::Tridiagonal{T}, i::Integer, j::Integer)
 end
 
 function setindex!(A::Tridiagonal, x, i::Integer, j::Integer)
+    @boundscheck checkbounds(A, i, j)
     if i == j
-        A.d[i] = x
+        @inbounds A.d[i] = x
     elseif i - j == 1
-        A.dl[j] = x
+        @inbounds A.dl[j] = x
     elseif j - i == 1
-        A.du[i] = x
-    else
-        throw(ArgumentError("cannot set elements outside the sub, main, or super diagonals"))
+        @inbounds A.du[i] = x
+    elseif !iszero(x)
+        throw(ArgumentError(string("cannot set entry ($i, $j) off the ",
+            "main, sub, and super diagonals to a nonzero value ($x)")))
```

Comment:
Will change to "or". (For my benefit, could you explain why "or" strikes you as more correct? Present understanding: "or" seems more correct in e.g. "entry not on a, b, or c", but "and" more correct in "entry off a, b, and c", as an "entry off a, b, or c" could be e.g. off a but on b and c? Thanks!)